tests/slsa_analyzer/test_git_url.py

```diff
@@ -122,36 +116,28 @@ def test_get_remote_vcs_url() -> None:
     assert git_url.get_remote_vcs_url("git@github.com:7999/org/") == ""
 
 
-def test_invalid_git_allowed_hosts() -> None:
+@patch("macaron.slsa_analyzer.git_url.get_allowed_git_service_domains")
+def test_get_remote_vcs_url_with_invalid_allowed_domains(
+    get_allowed_domains_fn: MagicMock,
+) -> None:
     """Test the vcs URL validator method without any allowed git hosts."""
-    original_allowed_hosts = defaults.get("git", "allowed_hosts")
-
-    # Test running with no git hosts defined in defaults.ini
-    defaults.remove_option("git", "allowed_hosts")
+    get_allowed_domains_fn.return_value = []
```

Comment:
Instead of mocking this function, wouldn't it be better to use a real `defaults` object and set invalid values? That way we can make sure that `git_url.get_allowed_git_service_domains` also works as expected.

Reply:
Hmm, I can probably create a separate test for `git_url.get_allowed_git_service_domains`?
Modifying the global `defaults` object looks quite ugly to me, and I'll have to make sure I revert it correctly on top of that 😄.
I also thought about mocking the `defaults` object itself, but it is also quite ugly.

Reply:
You can also read read a `defaults.ini` that has the test values and by merge the configurations using `defaults.load_defaults( "my_defaults.ini"))`.

Reply:
I added a couple of tests for `get_allowed_git_service_domains` in 2e607da3aa4f5bc3714940469c9e19bea566b14b.